src/main/java/teammates/ui/controller/InstructorFeedbackResultsDownloadAction.java

```diff
@@ -35,6 +38,11 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         String fileContent;
         String fileName;
 
+        // initialize SectionDetail correctly
+        if (section != null && sectionDetailValue != null && !sectionDetailValue.isEmpty()) {
+            sectionDetail = SectionDetail.valueOf(sectionDetailValue);
```

Comment:
Need to add a guard if `sectionDetailValue` is invalid.

Reply:
Guard as invalid meaning? Currently `selectedSectionDetail` will be default if the field `sectionDetailValue` is null. Do you mean checking if the `sectionDetailValue` has the same value as one of the enum shown in `SectionDetail.java`?

Reply:
> Do you mean checking if the `sectionDetailValue` has the same value as one of the enum shown in `SectionDetail.java`?

This is what I meant.